core/src/main/scala/shapeless/package.scala

```diff
@@ -59,6 +59,60 @@ package object shapeless {
   type ∃[P[_]] = P[T] forSome { type T }
   type ∀[P[_]] = ¬[∃[({ type λ[X] = ¬[P[X]]})#λ]]
 
+  //Type logic-gates
+  trait ||[A, B] extends Serializable {
+    def a: Option[A]
+    def b: Option[B]
+  }
+
```

Comment:
I don't think there was a particular reason, when writing it within our own code base it just fell out that we wrote it as `ev.a.isDefined`. I also originally didn't see a reason to bias the disjunction to the left or the right in the case of the equality. I can change it to `Either[A, B]`.